keras_nlp/layers/positional_embedding.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Positional embedding implementation based on `keras.layers.Layer`."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+class PositionalEmbedding(keras.layers.Layer):
+    """Creates a positional embedding.
+
+    Args:
+      max_length: The maximum size of the dynamic sequence.
+      initializer: The initializer to use for the embedding weights. Defaults to
+        "glorot_uniform".
+      seq_axis: The axis of the input tensor where we add the embeddings.
+
+    Example:
+    ```python
+    positional_embedding = PositionalEmbedding(max_length=100)
+    inputs = keras.Input((100, 32), dtype=tf.float32)
+    outputs = positional_embedding(inputs)
+    ```
+
+    Reference:
+      [BERT: Pre-training of Deep Bidirectional Transformers for Language
+      Understanding](https://arxiv.org/abs/1810.04805).
+    """
+
+    def __init__(
+        self, max_length, initializer="glorot_uniform", seq_axis=1, **kwargs
+    ):
+
+        super().__init__(**kwargs)
+        if max_length is None:
+            raise ValueError("`max_length` must be an Integer, not `None`.")
+        self.max_length = max_length
```

Comment:
Consider casting to `int()` to improve safety / consistency

Reply:
Will add